psi_cardinality/go/server/server.go

```diff
@@ -0,0 +1,135 @@
+package server
+
+/*
+#include "psi_cardinality/c/psi_cardinality_server_c.h"
+*/
+import "C"
+import (
+	"errors"
+	"fmt"
+	"runtime"
+)
+
+//PSICardinalityServer context
+type PSICardinalityServer struct {
+	context C.psi_cardinality_server_ctx
+}
+
+//CreateWithNewKey returns a new PSI server
+func CreateWithNewKey() (*PSICardinalityServer, error) {
+	psiServer := new(PSICardinalityServer)
+
+	var err *C.char
+	ret := C.psi_cardinality_server_create_with_new_key(&psiServer.context, &err)
+	if ret != 0 {
+		return nil, fmt.Errorf("failed to create server context: %v(%v)", C.GoString(err), ret)
+	}
+	if psiServer.context == nil {
+		return nil, errors.New("failed to create server context: null")
```

Comment:
Same as in Client, please make explicit if this indicates an implementation error on our side.